subsys/testsuite/ztest/include/zephyr/ztest_assert.h

```diff
@@ -282,6 +282,13 @@ static inline bool z_zexpect(bool cond, const char *default_msg, const char *fil
  */
 #define zassert_ok(cond, ...) zassert(!(cond), #cond " is non-zero", ##__VA_ARGS__)
 
+/**
+ * @brief Assert that @a cond is not 0 (failure)
+ * @param cond Condition to check
+ * @param ... Optional message and variables to print if the assertion fails
+ */
+#define zassert_nok(cond, ...) zassert(!!(cond), #cond " is zero", ##__VA_ARGS__)
```

Comment:
As an **actually**-blocking comment, can we spell out `not` instead of just `n`? As you pointed out, we already have `zassert_not_equals()`, so I'd rather keep consistent with the existing pattern. It's also possible to miss the single `n` when someone is in a rush or has lack of sleep.